### PR TITLE
[test] Build DBConfig from env var secrets

### LINKED ISSUE #6604: [test] 9 tests fail with "error parsing DB_PORT" for self-hosted
### Bug description

We'll need to [extend the test framework](https://github.com/gitpod-io/gitpod/blob/a9dca2a2bf7f1562ee90e53b038d63f9cbba824f/test/pkg/integration/apis.go#L432) as the DB port comes from a secret now.

### Steps to reproduce

[Follow these steps](https://github.com/gitpod-io/gitpod/tree/main/test#manually-against-a-kubernetes-cluster) against a self-hosted installation of Gitpod.

### Workspace affected

n/a

### Expected behavior

The tests should pass.

### Example repository

_No response_

### Anything else?

Will require a cluster in GCP to do testing. 

1 test was failing due to `minio.default.svc.cluster.local: no such host`. This may have been resolved in a recent commit, but double check. If not, see if it can be resolved.

4 tests were failing (even with the saas) due to `cannot start workspace`. This may have been resolved in a recent commit, but double check.

The failures were found in #6518.

### PR DESCRIPTION
**Pull request by kylos101**
## Description

Update integration tests to connect to database for self-hosted envs that used the installer (it uses a secret to store database connectivity details, which are made accessible through a secretRef as ENV VARs).

## Related Issue(s)

Fixes #6604

## How to test

[Run this and have tests pass.](https://github.com/gitpod-io/gitpod/tree/main/test#manually-against-a-kubernetes-cluster)

## Release Notes

```release-note
NONE
```

### CODE REVIEW

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **csweichel** after the PR has been reviewed.
You can assign the PR to them by writing `/assign @csweichel` in a comment when ready.

Associated issue: *#6604*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/gitpod-io/gitpod/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by kylos101** [comment]
@aledbf below you will find test results for this PR. Let me know what you think? 

A couple thoughts:
1. There are **new** failures for integration tests done against self hosted (tests which pass in staging-main for core-dev). I plan to create a new issue.
2. When branches main and kyleb/fixes-6604 are run against staging-main namespace on core-dev, the results are similar, which is good.

################################################
SELF HOSTED RESULTS (with branch kyleb/fixes-6604)

real    10m6.594s
user    0m51.641s
sys     0m18.503s

```
    content-service_test.go:275: HTTP PUT request failed: ... dial tcp: lookup minio.default.svc.cluster.local: no such host"
--- FAIL: TestUploadDownloadBlob (2.18s) **new**	

    content-service_test.go:275: HTTP PUT request failed: ... dial tcp: lookup minio.default.svc.cluster.local: no such host"
--- FAIL: TestUploadDownloadBlobViaServer (4.42s)  **new**

    builder_test.go:84: image build failed: image build did not produce a workspace image
--- FAIL: TestBaseImageBuild (173.81s)  **new**

    builder_test.go:194: image builder error: rpc error: code = DeadlineExceeded desc = context deadline exceeded
--- FAIL: TestParallelBaseImageBuild (300.01s) **existing**

    blockuser_test.go:71: cannot perform AdminBlockUser: "jsonrpc2: code 555 message: enterprise license required"
--- FAIL: TestAdminBlockUser (17.53s)  **new**

    server_test.go:96: cannot get workspace: "cannot wait for workspace: \"context deadline exceeded\""
--- FAIL: TestStartWorkspace (300.42s)  **new**

    content_test.go:41: cannot start workspace: "workspace instance 181af11b-34c2-478e-8f1f-b196e0181fce has stopped"
--- FAIL: TestBackup (5.25s)  **new**

    content_test.go:151: cannot start workspace: "cannot wait for workspace: \"context deadline exceeded\""
--- FAIL: TestMissingBackup (300.01s) **existing**

    prebuild_test.go:40: cannot launch a workspace: "cannot start workspace: \"workspace instance 151ca5f3-4739-4af6-b623-0ceeaebfc135 is stopping\""
--- FAIL: TestPrebuildWorkspaceTaskSuccess (32.05s) **existing**

    contexts_test.go:132: timed out attempting to connect agent: dial-http tcp localhost:43853: read tcp 127.0.0.1:52724->127.0.0.1:43853: read: connection reset by peer
--- FAIL: TestGitHubContexts (131.32s)  **new**

    contexts_test.go:121: cannot start workspace: "cannot wait for workspace: \"context deadline exceeded\""
--- FAIL: TestGitLabContexts (300.29s) **existing**

    docker_test.go:33: cannot start workspace: "workspace instance a803f619-3957-4dfc-9cc3-d7caf7cb5916 has stopped"
--- FAIL: TestRunDocker (8.72s) **existing**
```
	
################################################
core-dev results for staging-main namespace (with main branch)

real    10m40.074s
user    11m58.684s
sys     1m30.679s

```
    builder_test.go:194: image builder error: rpc error: code = DeadlineExceeded desc = context deadline exceeded
--- FAIL: TestParallelBaseImageBuild (300.01s)

    content_test.go:151: cannot start workspace: "cannot wait for workspace: \"context deadline exceeded\""
--- FAIL: TestMissingBackup (300.00s)

    prebuild_test.go:40: cannot launch a workspace: "cannot start workspace: \"workspace instance 5c73e042-218c-4d60-a0fa-97219b7865b2 is stopping\""
--- FAIL: TestPrebuildWorkspaceTaskSuccess (87.64s)

    contexts_test.go:121: cannot start workspace: "jsonrpc2: code 401 message: Unauthorized"
--- FAIL: TestGitLabContexts (4.41s)

    docker_test.go:33: cannot start workspace: "cannot wait for workspace: \"context deadline exceeded\""
--- FAIL: TestRunDocker (300.01s)
```

################################################
core-dev results for staging-main namespace (with branch kyleb/fixes-6604)

real    10m8.186s
user    0m56.330s
sys     0m19.745s

```
    builder_test.go:194: image builder error: rpc error: code = DeadlineExceeded desc = context deadline exceeded
--- FAIL: TestParallelBaseImageBuild (300.00s)

    content_test.go:151: cannot start workspace: "cannot wait for workspace: \"context deadline exceeded\""
--- FAIL: TestMissingBackup (300.00s)

    contexts_test.go:121: cannot start workspace: "jsonrpc2: code 401 message: Unauthorized"
--- FAIL: TestGitLabContexts (4.34s)

    docker_test.go:33: cannot start workspace: "cannot wait for workspace: \"context deadline exceeded\""
--- FAIL: TestRunDocker (300.01s)
```

File: test/pkg/integration/apis.go [hunk 1]
```diff
@@ -520,22 +520,27 @@ func FindDBConfigFromPodEnv(componentName string, namespace string, client klien
...
 		for _, v := range c.Env {
+			var findErr error
 			if v.Name == "DB_PASSWORD" {
-				password = v.Value
+				password, findErr = FindValueFromEnvVar(v, client, namespace)
 			} else if v.Name == "DB_PORT" {
-				pPort, err := strconv.ParseUint(v.Value, 10, 16)
+				var portStr string
+				portStr, findErr = FindValueFromEnvVar(v, client, namespace)
+				pPort, err := strconv.ParseUint(portStr, 10, 16)
```
Comment (csweichel) [inline test/pkg/integration/apis.go:534]: edge case: if `findErr != nil` it's likely that `password == ""` when will make `ParseUint` fail. In that case the "error parsing DB_PORT" error below would shadow `findErr`.
Possible solution: check for `findErr` immediately after making the call.

Comment (kylos101) [reply test/pkg/integration/apis.go:540]: Good eye, thank you! That seems like the best practice (immediately handle errors). Resolved in 475ef294fdb560756bd623f02c46df019dac9396.

**Comment by roboquat** [comment]
LGTM label has been added.  <details>Git tree hash: ca60cb3861e8892d66420133f995ad57d1150504</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/6659#issuecomment-966987520" title="LGTM">csweichel</a>*

Associated issue: *#6604*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/gitpod-io/gitpod/blob/main/OWNERS)~~ [csweichel]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>
